src/asqi/validation.py

```diff
@@ -2,12 +2,91 @@
 from pathlib import Path
 from typing import Any, Dict, List, Optional
 
+import yaml
 from pydantic import BaseModel
+from rich.console import Console
 
+from asqi.config import load_config_file
+from asqi.errors import DuplicateTestIDError
 from asqi.schemas import Manifest, SuiteConfig, SystemsConfig
 
 logger = logging.getLogger()
 
+# Initialize Rich console and execution queue
+console = Console()
+
+
+def validate_test_ids(test_suite_config_path: str) -> None:
+    """
+    Validates that all test IDs in the suite file are unique.
+
+    Args:
+        test_suite_config_path: Path to the test suite file
+
+    Raises:
+        DuplicateTestIDError: If duplicate test IDs are found, with detailed information about all duplicate locations
+    """
+    all_test_ids: Dict[str, List[str]] = {}
+
+    if test_suite_config_path:
+        try:
+            suite_config = load_config_file(test_suite_config_path)
+            extract_suite_ids(all_test_ids, suite_config, test_suite_config_path)
+
+        except yaml.YAMLError as e:
+            console.print(
+                f"[yellow]ID Validation. Failed to parse test suite file {test_suite_config_path}:[/yellow] {e}"
+            )
+        except Exception as e:
```

Comment:
If there are errors over here, we should raise it instead of printing to console?